wgpu-core/src/command/render.rs

```diff
@@ -986,7 +955,7 @@ impl<'d> RenderPassInfo<'d> {
             let ds_aspects = view.desc.aspects();
 
             if !ds_aspects.contains(hal::FormatAspects::STENCIL)
-                || (at.stencil.load_op() == at.depth.load_op()
+                || (at.stencil.load_op().discriminant() == at.depth.load_op().discriminant()
```

Comment:
Can we use `matches!` instead?

Reply:
The problem is that `LoadOp::Clear` has data, that can't match. I am not aware of any way to write that with patterns for `matches!`.

Reply:
It can be done but inlining it would probably be too much.

```rust
matches!(
    (at.stencil.load_op(), at.depth.load_op()),
    (LoadOp::Clear(_), LoadOp::Clear(_)) | (LoadOp::Load, LoadOp::Load)
)
```

Maybe we should add a function with the match instead?

```rust
match (at.stencil.load_op(), at.depth.load_op()) {
    (LoadOp::Clear(_), LoadOp::Clear(_)) => true,
    (LoadOp::Load, LoadOp::Load) => true,
    _ => false,
}
```

Ultimately the `discriminant` method works but we probably don't need/want these enums to be `#[repr(u8)]`/`#[repr(C)]`. That would make other things non `#[repr(C)]` but that's fine; we've moved in this direction overall (https://github.com/gfx-rs/wgpu/pull/6624), binding crates can come up with a C compatible API.

Reply:
Actually repr is mostly because it was present on previous LoadOp/StoreOp and I didn't want to be breaking too much.

Instead of doing matches! directly I packed into a method on LoadOp.